src/store/actions/auth.js

```diff
@@ -130,13 +162,104 @@ export const authGo = (email, password, isRegister) => {
   }
 };
 
+export const authSetProfileGo = (token, displayName, photoUrl, isEdit) => {
+  return (dispatch) => {
+    const apiKey = process.env.REACT_APP_FIREBASE_API_KEY;
+
+    const authData = {
+      idToken: token,
+      displayName: displayName,
+      photoUrl: photoUrl
+    }
+
+    let url = `https://identitytoolkit.googleapis.com/v1/accounts:update?key=${apiKey}`;
+
+    axios.post(url, authData)
+      .then((response) => {
+        dispatch(authDoneLoading());
+        dispatch(closeModal());
+        setTimeout(() => {
+          dispatch(openAndSetActiveModalAndMessage('success', `Your profile has been ${isEdit ? 'updated' : 'set'}!`));
+        }, 250);
+        dispatch(authGetProfile(token));
+      }).catch((error) => {
+        let errorMessage = '';
+        if(error.response) {
+          errorMessage = normalizeErrorString(error.response.data.error.message);
+        } else {
+          errorMessage = error.message;
+        }
+
+        dispatch(authFail(errorMessage));
+        dispatch(closeModal());
+        setTimeout(() => {
+          dispatch(openAndSetActiveModalAndMessage('error', errorMessage));
+        }, 250);
+      });
+  }
+}
+
+export const authSetProfile = (token, displayName, photoUrl, isEdit) => {
+  return (dispatch) => {
+    dispatch(authStart());
+
+    const photoUpdated = typeof(photoUrl) === 'object';
+    if(photoUpdated) {
+      const ref = firebaseStorageRef.child(photoUrl.name);
+
+      ref.put(photoUrl).then((response) => {
+        ref.getDownloadURL().then((response) => {
+          dispatch(authSetProfileGo(token, displayName, response, isEdit));
+        });
+      });
+    } else {
+      dispatch(authSetProfileGo(token, displayName, photoUrl, isEdit));
+    }
+  }
+};
+
+export const authGetProfile = (token) => {
+  return (dispatch) => {
+    dispatch(authGetProfileStart());
+
+    const apiKey = process.env.REACT_APP_FIREBASE_API_KEY;
+
+    let url = `https://identitytoolkit.googleapis.com/v1/accounts:lookup?key=${apiKey}`;
+
+    const authData = {
+      idToken: token
+    }
+
+    axios.post(url, authData)
+      .then((response) => {
+        const { displayName, photoUrl, email } = response.data.users[0];
+
+        dispatch(authGetProfileSuccess(displayName, photoUrl, email));
+      }).catch((error) => {
+        let errorMessage = '';
```

Comment:
Another ternary candidate... also, it seems that this could even be a common helper function (the if..else block, I mean, which sets the `errorMessage)

Reply:
Right, I was thinking I had that written out more than once